core/include/scipp/core/element/comparison.h

```diff
@@ -34,12 +43,17 @@ constexpr auto isclose_units = [](const units::Unit &x, const units::Unit &y,
 constexpr auto isclose = overloaded{
     transform_flags::expect_no_variance_arg_t<2>{}, isclose_types_t{},
     isclose_units, [](const auto &x, const auto &y, const auto &t) {
-      using std::abs;
-      return abs(x - y) <= t;
+      if constexpr (std::is_same_v<std::decay_t<decltype(x)>,
+                                   Eigen::Vector3d>) {
+        return (x - y).norm() <= t;
+      } else {
+        using std::abs;
+        return abs(x - y) <= t;
+      }
     }};
 
 constexpr auto isclose_equal_nan = overloaded{
-    transform_flags::expect_no_variance_arg_t<2>{}, isclose_types_t{},
+    transform_flags::expect_no_variance_arg_t<2>{}, isclose_common_types_t{},
```

Comment:
I suppose we *can* have NAN also as vector elements? Should we support this here then?

Reply:
I was mainly avoiding handling to get a first draft ready, but agree, it would be useful to users to have this given that NAN is a possibility.

Reply:
Maybe the question is resolved auotmatically by my new comment below?